src/main/java/io/cryostat/rules/RuleProcessor.java

```diff
@@ -154,8 +154,10 @@ public void onEvent(Event<RuleEvent, Rule> event) {
                                                 .filter(
                                                         serviceRef ->
                                                                 registry.applies(
-                                                                        event.getPayload(),
-                                                                        serviceRef))
+                                                                                event.getPayload(),
+                                                                                serviceRef)
+                                                                        && event.getPayload()
```

Comment:
Let's put the `isEnabled()` check before the `applies()` check. That way if the rule is disabled we can skip evaluating the match expression and save some CPU time.